addons/IWB_Toolbar/src/presenter.js

```diff
@@ -2002,36 +2002,40 @@ function AddonIWB_Toolbar_create() {
             lastScrollTop = 0,
             panelTop = 0;
         if (!presenter.playerController || presenter.playerController.isPlayerInCrossDomain()) return;
-
+        var $defaultScrollElement = $(window.parent.document);
+        var $mCourserScrollElement = $defaultScrollElement.find('#lesson-view > div > div');
+        var scrollElements = [$defaultScrollElement, $mCourserScrollElement];
         try {
-            $(window.parent.document).scroll(function () {
-                if (presenter.isOnScreen(presenter.$view.parent(), window)) {
-                    var containerHeight = presenter.$pagePanel.outerHeight(true),
-                        scrollTop = $(this).scrollTop(),
-                        min = presenter.$pagePanel.offset().top,
-                        headerHeight = $('.ic_header').outerHeight(true) - 20,
-                        max = containerHeight + headerHeight,
-                        zoomHeightScale = presenter.getZoomHeightScale();
-                    difference = scrollTop - lastScrollTop;
-
-                    panelTop = parseInt(presenter.$panel.css('top'), 10) + difference * zoomHeightScale;
-                    lastScrollTop = scrollTop;
-
-                    if (panelTop && (panelTop) > min && (panelTop) < max) {
-                        presenter.$panel.css({
-                            'top': (panelTop) + 'px'
-                        });
-                    } else if (panelTop && (panelTop) >= max) {
-                        presenter.$panel.css({
-                            'top': (containerHeight - presenter.$panel.outerHeight(true) + min) + 'px'
-                        });
-                    } else if (panelTop && (panelTop) <= min) {
-                        presenter.$panel.css({
-                            'top': min + 'px'
-                        });
+            for (var i = 0; i < scrollElements.length; i++) {
```

Comment:
Wydaje mi się, że nie powinno to się wykonywać dwa razy. Czyli, jeżeli znajdzie $mCourserScrollElement to nie wykonuje już $defaultScrollElement

Reply:
Nie masz gwarancji, że mCourserScrollElement to będzie akurat ten, na którym jest scrollowanie. I tak tylko jeden callback będzie wywoływany, bo tylko na jednym będzie scrollowanie.

Reply:
OK